src/fides/api/alembic/migrations/helpers/fideslang_migration_functions.py

```diff
@@ -0,0 +1,451 @@
+            update_label_query,
+            {
+                "key": row["provided_identity_id"],
+                "old_use": row["data_use"],
+                "updated_label": updated_use,
+            },
+        )
+
+    # Update the Privacy Request Table
+    existing_privacy_requests: ResultProxy = bind.execute(
+        text("select id, consent_preferences from privacyrequest;")
+    )
+
+    for row in existing_privacy_requests:
+        preferences: List[Dict] = row["consent_preferences"]
+
+        if preferences:
+            for index, preference in enumerate(preferences):
+                preferences[index]["data_use"] = _replace_matching_data_label(
+                    data_label=preference["data_use"], data_label_map=data_use_map
+                )
+
+        update_pr_query: TextClause = text(
+            "UPDATE privacyrequest SET consent_preferences= :updated_preferences WHERE id= :id"
+        )
+        bind.execute(
+            update_pr_query,
+            {"id": row["id"], "updated_preferences": json.dumps(preferences)},
+        )
+
+    # Update the Consent Request Table
+    existing_consent_requests: ResultProxy = bind.execute(
+        text("select id, preferences from consentrequest;")
+    )
+
+    for row in existing_consent_requests:
+        preferences: List[Dict] = row["preferences"]
+
+        if preferences:
+            for index, preference in enumerate(preferences):
+                preferences[index]["data_use"] = _replace_matching_data_label(
+                    data_label=preference["data_use"], data_label_map=data_use_map
+                )
+
+        update_cr_query: TextClause = text(
+            "UPDATE consentrequest SET preferences= :updated_preferences WHERE id= :id"
+        )
+        bind.execute(
+            update_cr_query,
+            {"id": row["id"], "updated_preferences": json.dumps(preferences)},
+        )
+
+
+def remove_conflicting_rule_targets(db: Session):
+    """
+    Iterates through all of the erasure policies and removes level 3 data categories in favor of level 2 data categories.
+
+    For example: user.demographic is preserved over user.demographic.*
+    """
+    erasure_rules = Rule.filter(
+        db=db, conditions=(Rule.action_type == ActionType.erasure)
+    ).all()
+
+    for rule in erasure_rules:
+        all_categories = {target.data_category for target in rule.targets}
+
+        rule_targets_to_remove = []
+        for target in rule.targets:
+            parts = target.data_category.split(".")
+            if len(parts) == 3:
+                parent_category = f"{parts[0]}.{parts[1]}"
+                if parent_category in all_categories:
+                    rule_targets_to_remove.append(target)
+
+        for rule_target in rule_targets_to_remove:
+            logger.info(
+                f"Removing conflicting rule target {rule_target.data_category} for rule {rule.key}"
+            )
+            db.delete(rule_target)
+
+    db.commit()
```

Comment:
I'd try to avoid using this commit()-  it shouldn't be needed, or should be rewritten so it isn't needed - see the other functions in this file.

I think the way we run migrations, it's all wrapped in a single transaction, so all migration scripts are run as part of one transaction and the commit happens at the end.  We've had issues where say we're running migrations and something fails or takes too long and we have to bail in the middle - it's pretty straightforward to stop and roll back because nothing was actually committed.  We could get in a weird state by deviating from this pattern where we get some of the migration committed but not all of it in the event of a failure.